### PR TITLE
fix: OwnerFreezable timestamp boundaries inclusive to match certifiedUntil (#296)

### LINKED ISSUE #296: Internal inconsistency: OwnerFreezable timestamp boundaries are exclusive, certifiedUntil is inclusive
Within rain.vats itself, the timestamp boundary convention is inconsistent:

| Location | Comparison | Semantic |
|---|---|---|
| `OffchainAssetReceiptVault.sol:608` `certifiedUntil` | `block.timestamp > certifiedUntil → expired` | Valid through and including (inclusive) |
| `OwnerFreezable.sol:142,178,196` `alwaysAllowedFroms` / `ownerFrozenUntil` | `block.timestamp < deadline → allowed/frozen` | Strictly before deadline (exclusive) |

Two different conventions in the same package, used by the same vault.

Consumers building on top end up forced to pick one and inherit the other implicitly. Mixing them on the same timestamp `T` produces asymmetric behaviour at exactly `block.timestamp == T`: an `OwnerFreezable` allowance with deadline `T` is already inactive, while anything keyed off `certifiedUntil == T` is still valid.

Solady's `TimedRoles` (`start <= block.timestamp <= expires`) is inclusive on both endpoints — the de-facto convention in EVM ecosystem code where it's been thought through.

### Desired change

Pick inclusive (`<=` / `>=`) for `OwnerFreezable` to match `certifiedUntil`. Single internally-consistent convention across rain.vats.

Breaking — coordinated version bump + consumer review.

### PR DESCRIPTION
**Pull request by thedavidmeister**
Fixes #296.

Switch the three `OwnerFreezable` boundary comparisons from `<` to `<=` so the package is internally consistent with `OffchainAssetReceiptVault.certifiedUntil` (also inclusive). Without this, consumers mixing the two on the same timestamp see asymmetric behaviour at the exact boundary block — the freeze releases while certification is still valid (or vice versa, depending on which they pin to).

## Source changes

- `ownerFreezeStopAlwaysAllowingFrom` / `ownerFreezeStopAlwaysAllowingTo`: protection active through and including `protectedUntil`; removable only at `block.timestamp > protectedUntil`.
- `ownerFreezeCheckTransaction`: freeze active through and including `ownerFrozenUntil`.
- Storage field NatSpec updated to describe the inclusive semantics directly.

## Test updates

- `checkOwnerFreezeStopAllowing*` helpers warp to `protectedUntil + 1`, with `vm.assume(protectedUntil < type(uint256).max)` because permanent protection (deadline at max uint256) has no later block to remove from.
- `testOwnerFreezable*ProtectedReverts` bound the revert range to `[0, protectUntil]` inclusive.
- `giveReasonToTransfer` (vault test) warps to `frozenUntil + 1`.
- `setupTokenTransferTest` certifies to `type(uint128).max` so certification outlasts the freeze warp (was `block.timestamp + 1` which now expires at the same warp under inclusive semantics).

## Test plan
- [x] full suite passes (329/329)
- [ ] CI green

## Summary by CodeRabbit

* **Bug Fixes**
  * Refined owner freeze/protection boundary conditions to use inclusive timestamp semantics. Restrictions now apply through the exact `block.timestamp` boundary rather than stopping immediately before it.

* **Tests**
  * Added comprehensive boundary-pinning tests to verify correct behavior at exact freeze/protection timeout transitions.

### CODE REVIEW

File: src/abstract/OwnerFreezable.sol [hunk 1]
```diff
@@ -137,9 +140,12 @@ abstract contract OwnerFreezable is IOwnerFreezableV1, OwnableUpgradeable {
...
-        // If the current time is after the protection for this `from` then
-        // we can remove it. Otherwise we revert to respect the protection.
-        if (block.timestamp < s.alwaysAllowedFroms[from]) {
+        // Inclusive boundary: protection is active through and including
+        // `protectedUntil`. Removal becomes possible only once
+        // `block.timestamp > protectedUntil`. Matches the convention used
+        // by `OffchainAssetReceiptVault.certifiedUntil` and the corporate
+        // action `effectiveTime` semantics on consumers.
+        if (block.timestamp <= s.alwaysAllowedFroms[from]) {
             revert OwnerFreezeAlwaysAllowedFromProtected(from, s.alwaysAllowedFroms[from]);
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply src/abstract/OwnerFreezable.sol:179]: Addressed in 0622aea — but at init time rather than per-call. Added `__OwnerFreezable_init()` that reverts with `CorruptedEnvironmentBlockTimestampZero` if the inheritor's initializer runs at `block.timestamp == 0`. Refusing the deployment at the boundary keeps the per-call comparisons clean (`block.timestamp <= deadline` everywhere, matching `certifiedUntil`) and treats the genesis-zero state as the environment corruption it actually is rather than a value the runtime checks need to disambiguate.
